addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -165,6 +176,59 @@ export class PosStore extends Reactive {
         return !this.cashier ? "LoginScreen" : "ProductScreen";
     }
 
+    setActivityListeners() {
+        IDLE_TIMER_SETTER = this.setIdleTimer.bind(this);
+        for (const event of NON_IDLE_EVENTS) {
+            window.addEventListener(event, IDLE_TIMER_SETTER);
+        }
+    }
+
+    shouldResetIdleTimer() {
+        const stayPaymentScreen =
+            this.mainScreen.component === PaymentScreen && this.get_order().payment_ids.length > 0;
+        return !stayPaymentScreen;
+    }
+
+    setIdleTimer() {
```

Comment:
Replace the whole method by the one I sent you in the previous review

Reply:
It is updated now